### PR TITLE
Add support for aws_ec2_vpc_endpoint

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary: 
22de7047 Added integration test for 05_aws_ec2_vpc_endpoint
892d0ed2 Updated documentation and examples for aws-ec2-vpc-endpoint
f00ca905 Updated playbooks in aws role to support aws_ec2_vpc_endpoint
cf7db20c Updated schema to support aws_ec2_vpc_endpoint
7439ab69 Update requirements.txt for compatible boto version

### CODE REVIEW

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

File: config/Dockerfiles/tests.d/aws/05_aws_ec2_vpc_endpoint [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    set +e
+    linchpin -w . -v destroy "${TARGET}"
+}
+trap clean_up EXIT SIGHUP SIGINT SIGTERM
+
+pushd docs/source/examples/workspaces/${PROVIDER}
+
+echo "add default profile in config"
+mkdir -p ~/.aws/
+printf "[default]\nregion = us-east-1" > ~/.aws/config
```
Comment (p3ck) [inline config/Dockerfiles/tests.d/aws/05_aws_ec2_vpc_endpoint]: Just curious why we have to do this?

Comment (samvarankashyap) [reply config/Dockerfiles/tests.d/aws/05_aws_ec2_vpc_endpoint]: @p3ck : I was getting the following [error](https://stackoverflow.com/questions/34134879/aws-the-config-profile-myname-could-not-be-found )
aws_ec2_vpc_endpoint was looking for a default profile in the config . I had to create that to resolve the error.

Comment (p3ck) [reply config/Dockerfiles/tests.d/aws/05_aws_ec2_vpc_endpoint]: We don't need this for the other aws providers?  Does this need to be added to this providers documentation in linchpin?

Comment (samvarankashyap) [reply config/Dockerfiles/tests.d/aws/05_aws_ec2_vpc_endpoint]: No we donot need this for other modules . However, In my local testing I tried removing the file and it was working fine. I believe it is specific to this module in our container environment.

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
@p3ck @14rcole review required

File: config/Dockerfiles/tests.d/aws/02_aws_ec2_vpc_net [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash -xe
 
 # Verify the aws ec2 vpc net provisioning
-# distros.exclude: centos7 fedora28
+# distros.exclude: centos7 fedora27 fedora28
```
Comment (p3ck) [inline config/Dockerfiles/tests.d/aws/02_aws_ec2_vpc_net]: Shouldn't this be 28 and 29 now?

Comment (p3ck) [reply config/Dockerfiles/tests.d/aws/02_aws_ec2_vpc_net]: Probably needs to be rebased?

Comment (samvarankashyap) [reply config/Dockerfiles/tests.d/aws/02_aws_ec2_vpc_net]: It should be just 28 for now since I would like to test it on f29 only.

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**